jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +65,38 @@ def check_referer(self):
             return False
         return True
 
+    @functools.lru_cache
```

Comment:
This is one of those weird decorators where you have to call the function to return the decorator itself:

```suggestion
    @functools.lru_cache()
```

Reply:
Oh I missed that. I took the example from here: https://docs.python.org/3/library/functools.html where they don't use parentheses. What is the difference between that situation and this one?